### PR TITLE
Setup publishing the artifacts to the Bintray maven repository

### PR DESCRIPTION
**Pull request by lamtev**
* Setup publishing the artifacts to the Bintray maven repository
* Make `jar` Gradle task assemble a Jar with all classes included
* Remove all unnecessary dependencies
* Update README with instructions on how to use all this stuff

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -1,61 +1,74 @@
...
 }
 
 dependencies {
-    compile "org.jetbrains.kotlin:kotlin-reflect:$kotlin_version"
-    compile "org.jetbrains.kotlin:kotlin-stdlib-jre8:$kotlin_version"
-    compile fileTree(dir: 'lib', include: '*.jar')
-    compileOnly "org.projectlombok:lombok:1.18.12"
+    implementation "org.jetbrains.kotlin:kotlin-reflect:1.3.72"
+    implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk8:1.3.72"
+    implementation files("lib/kovenant-core-3.3.0.jar")
```
Comment (VanyaKrylov) [inline build.gradle]: Remove redundant dependency

File: build.gradle [hunk 2]
```diff
@@ -1,61 +1,74 @@
...
+    implementation files("lib/suffixtree-0.1.jar")
+    implementation files("lib/union-find-set-1.0.jar")
```
Comment (VanyaKrylov) [inline build.gradle]: Redundant as well - remove

File: build.gradle [hunk 3]
```diff
@@ -1,61 +1,74 @@
...
 
 dependencies {
-    compile "org.jetbrains.kotlin:kotlin-reflect:$kotlin_version"
-    compile "org.jetbrains.kotlin:kotlin-stdlib-jre8:$kotlin_version"
-    compile fileTree(dir: 'lib', include: '*.jar')
-    compileOnly "org.projectlombok:lombok:1.18.12"
+    implementation "org.jetbrains.kotlin:kotlin-reflect:1.3.72"
+    implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk8:1.3.72"
+    implementation files("lib/kovenant-core-3.3.0.jar")
+    implementation files("lib/suffixtree-0.1.jar")
```
Comment (VanyaKrylov) [inline build.gradle]: Replace with `implementation("com.suhininalex:suffixtree:1.0.2")`. 
**UPD:** To locate the dependency, add `maven {
        url = uri("https://dl.bintray.com/vorpal-research/kotlin-maven/")
    }` into `repositories` section of `build.gradle` file

**Review by VanyaKrylov — CHANGES_REQUESTED** [review]

File: build.gradle (hunk 1 above)
Comment (lamtev) [reply build.gradle]: done

File: build.gradle (hunk 2 above)
Comment (lamtev) [reply build.gradle]: done

File: build.gradle (hunk 3 above)
Comment (lamtev) [reply build.gradle]: done

**Review by VanyaKrylov — APPROVED** [review]

**Pull request merged**